<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integration of ChIP-seq and RNA-seq data in R • targetShop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Integration of ChIP-seq and RNA-seq data in R">
<meta property="og:description" content="targetShop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">targetShop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workshop_code.html">Integration of ChIP-seq and RNA-seq data in R</a>
    </li>
    <li>
      <a href="../articles/workshop_quiz.html">Quiz: Integration of ChIP-seq and RNA-seq data in R</a>
    </li>
    <li>
      <a href="../articles/workshop_syllabus.html">Syllabus: Integration of ChIP-seq and RNA-seq data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MahShaaban/targetShop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_code_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="workshop_code_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="workshop_code_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integration of ChIP-seq and RNA-seq data in R</h1>
                        <h4 class="author">Mahmoud Ahmed<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
            <h4 class="date">2020-11-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/MahShaaban/targetShop/blob/master/vignettes/workshop_code.Rmd"><code>vignettes/workshop_code.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_code.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co"># load required libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org">readr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org">purrr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi">AnnotationDbi</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MahShaaban/target">target</a></span><span class="op">)</span></pre></div>
<div id="use-case" class="section level1">
<h1 class="hasAnchor">
<a href="#use-case" class="anchor"></a>Use Case</h1>
<p>YY1 and YY2 belongs to the same family of transcription factors. YY1 is a zinc finger protein which direct deacetylase and histone acetyltransferases of the promoters of many genes. The results of the binding of YY1 to the regulatory regions of genes is the induction or repression of their expression. YY2 is a parloge of YY1. Similarly, it is a zinc finger protein with both activation or repression functions on its targets. Using the target analysis, we will attempt to answer the following questions. Do the two transcription factors share the same target genes? What are the consequences of the binding of each factor on its targets? If the two factors share binding sites, what is the function of the binding of the two factor to these sites?</p>
<p>To answer these questions, we use publicly available datasets to model the binding and gene expression under the transcription factors perturbations (Table @ref(tab:datasets)). This dataset was obtained in the form of differential expression between the two conditions from <a href="http://www.licpathway.net/KnockTF/index.html">KnockTF</a>. The first dataset is geneexpression profiling using microarrays of YY1/YY2 knockdown and control HeLa cells. The binding sites of the factors in HeLa cells were determined using two ChIP-Seq datasets. The ChIP peaks were obtained in the form of bed files from <a href="https://chip-atlas.org">ChIP-Atlas</a>. Finally, we used the USSC hg19 human genome to extract the genomic annotations.</p>
<p>Briefly, we first prepared the three sources of data for the target analysis. Then we predict the specific targets for each individual factors. Third, we predict the combined function of the two factors on the shared target genes. Finally, we show an example of a motif analysis of the competitively and cooperatively regulated targets.</p>
<table class="table">
<caption>(#tab:datasets) Expression and binding data of YY1 and YY2 in HeLa cells.</caption>
<thead><tr class="header">
<th>GEO ID</th>
<th>Data Type</th>
<th>Design</th>
<th>Ref.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>GSE14964</td>
<td>Microarrays</td>
<td>YY#-knockdown</td>
<td><span class="citation">Chen et al. (2010)</span></td>
</tr>
<tr class="even">
<td>GSE31417</td>
<td>ChIP-Seq</td>
<td>YY1 vs input</td>
<td><span class="citation">Michaud et al. (2013)</span></td>
</tr>
<tr class="odd">
<td>GSE96878</td>
<td>ChIP-Seq</td>
<td>YY2 vs input</td>
<td><span class="citation">Wu et al. (2017)</span></td>
</tr>
</tbody>
</table>
<div id="preparing-the-binding-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-the-binding-data" class="anchor"></a>Preparing the binding data</h2>
<p>The ChIP peaks were downloaded in the form of separate bed files for each factor. We first locate the files in the <code>data/</code> directory and load the files using <code>import.bed</code>. Then the data is transformed into a suitable format, <code>GRanges</code>. The resulting object, <code>peaks</code>, is a <code>list</code> of two <code>GRanges</code> items, one for each factor.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># locate the peaks bed files</span>
<span class="va">peak_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>YY1 <span class="op">=</span> <span class="st">'data/Oth.Utr.05.YY1.AllCell.bed'</span>,
                YY2 <span class="op">=</span> <span class="st">'data/Oth.Utr.05.YY2.AllCell.bed'</span><span class="op">)</span>

<span class="co"># load the peaks bed files as GRanges</span>
<span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">peak_files</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html">import.bed</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></pre></div>
</div>
<div id="preparing-the-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-the-expression-data" class="anchor"></a>Preparing the expression data</h2>
<p>The differential expression data were downloaded in tabular format. After locating the files in <code>data/</code>, we read the files using <code>read_tsv</code> and select and rename the relevant columns. The resulting object, <code>express</code>, is a <code>list</code> of two <code>tibble</code> items.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co"># locate the expression text files</span>
<span class="va">expression_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>YY1 <span class="op">=</span> <span class="st">'data/DataSet_01_18.tsv'</span>,
                      YY2 <span class="op">=</span> <span class="st">'data/DataSet_01_19.tsv'</span><span class="op">)</span>

<span class="co"># load the expression text files</span>
<span class="va">express</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">expression_files</span>,
               <span class="op">~</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="va">.x</span>, col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">9</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co">#9</span>
                 <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'tf'</span>, <span class="st">'gene'</span>, <span class="st">'fc'</span>, <span class="st">'pvalue'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">tf</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'YY1'</span>, <span class="st">'YY2'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></pre></div>
<p>The knockdown of either factor in HeLa cells seem to change the expression of many genes in either directions (Figure @ref(fig:foldchange)A&amp;B). Moreover, the changes resulting from the knockdown of the factors individually are correlated (Figure @ref(fig:foldchange)C). This suggest that, many of the regulated genes are shared targets of the two factors or they respond similarly to their perturbation of either factor.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># Figure 1</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="co"># volcano plot of YY1 knockdown</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">fc</span>, 
     <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>,
     xlab <span class="op">=</span> <span class="st">'Fold-change (log_2)'</span>,
     ylab <span class="op">=</span> <span class="st">'P-value (-log_10)'</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(A)'</span><span class="op">)</span>

<span class="co"># volcano plot of YY2 knockdown</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">fc</span>, 
     <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>,
     xlab <span class="op">=</span> <span class="st">'Fold-change (log_2)'</span>,
     ylab <span class="op">=</span> <span class="st">'P-value (-log_10)'</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(B)'</span><span class="op">)</span>

<span class="co"># plot fold-change of YY1 and YY2</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span>,
     <span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span>,
     xlab <span class="op">=</span> <span class="st">'YY1-knockdown (log_2)'</span>,
     ylab <span class="op">=</span> <span class="st">'YY2-knockdown (log_2)'</span>,
     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(C)'</span><span class="op">)</span></pre></div>
<div class="figure">
<img src="workshop_code_files/figure-html/foldchange-1.png" alt="Differential expression between factor knockdown and control HeLa cells. Gene expression was compared between transcription factors knockdown and control HeLa cells. The fold-change and p-values of (A) YY1- and (B) YY2-knockdown are shown as volcano plots. (C) Scatter plot of the fold-change of the YY1- and YY2-knockdown." width="100%"><p class="caption">
Differential expression between factor knockdown and control HeLa cells. Gene expression was compared between transcription factors knockdown and control HeLa cells. The fold-change and p-values of (A) YY1- and (B) YY2-knockdown are shown as volcano plots. (C) Scatter plot of the fold-change of the YY1- and YY2-knockdown.
</p>
</div>
</div>
<div id="preparing-genome-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-genome-annotation" class="anchor"></a>Preparing genome annotation</h2>
<p>The gene information in <code>express</code> is recorded using the gene SYMBOLS. They need to be mapped to the ENTREZIDS before extracting the genomic coordinates. To do that, we use the <code>org.Hs.eg.db</code> to map between the identifiers. Next, we use the <code>TxDb.Hsapiens.UCSC.hg19.knownGene</code> to get the genomic coordinates for the transcripts and resize them to a 100kb upstream from the transcription start sites.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># load genome data</span>
<span class="va">symbol_entrez</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">express</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span>,
                        <span class="st">'ENTREZID'</span>, <span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'gene'</span>, <span class="st">'gene_id'</span><span class="op">)</span><span class="op">)</span>

<span class="co"># format genome to join with express</span>
<span class="va">genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">promoters</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
            upstream <span class="op">=</span> <span class="fl">100000</span>,
            columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'tx_id'</span>, <span class="st">'tx_name'</span>, <span class="st">'gene_id'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span><span class="op">)</span></pre></div>
<p>The resulting object, <code>genome</code>, from the previous step is a <code>tibble</code> that shares the column <code>gene_id</code> with the expression data <code>express</code>. Now the two objects can be merged. The merged object, <code>regions</code>, is similarly a <code>tibble</code> that contains genome and expression information of all common genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># make regions by merging the genome and express data</span>
<span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">express</span>,
               <span class="op">~</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">genome</span>, <span class="va">symbol_entrez</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span>
                 <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html">makeGRangesFromDataFrame</a></span><span class="op">(</span>keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></pre></div>
</div>
<div id="predicting-gene-targets-of-individual-factors" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-gene-targets-of-individual-factors" class="anchor"></a>Predicting gene targets of individual factors</h2>
<p>The standard target analysis includes the identification of associated peaks using <code>associated_peaks</code> and direct targets using <code>direct_targets</code>. The input for these functions are the objects <code>peaks</code> and <code>regions</code> from the previous steps in addition to the column names for regions <code>regions_col</code> or the region and the statistics column <code>stats_col</code> which is the fold-change in this case. The resulting objects are <code>GRanges</code> for the identified peaks assigned to the regions <code>ap</code> or the ranked targets. Several columns is added to the metadata objects of the <code>GRanges</code> to save the calculations.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># get associated peaks</span>
<span class="va">ap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="va">regions</span>,
           <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/target/man/associated_peaks.html">associated_peaks</a></span><span class="op">(</span>peaks<span class="op">=</span><span class="va">.x</span>,
                             regions <span class="op">=</span> <span class="va">.y</span>,
                             regions_col <span class="op">=</span> <span class="st">'tx_id'</span><span class="op">)</span><span class="op">)</span>
<span class="co"># get direct targets</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="va">regions</span>,
           <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/target/man/direct_targets.html">direct_targets</a></span><span class="op">(</span>peaks<span class="op">=</span><span class="va">.x</span>,
                           regions <span class="op">=</span> <span class="va">.y</span>,
                           regions_col <span class="op">=</span> <span class="st">'tx_id'</span>,
                           stats_col <span class="op">=</span> <span class="st">'fc'</span><span class="op">)</span><span class="op">)</span></pre></div>
<p>To determine the dominant function of a factor, we divide the targets by the direction of the effect of knockdown of the factor on the expression of the target and show the regulatory potential of the target on these groups. We use the empirical distribution function (ECDF) to show the fraction of targets targets at a specified regulatory potential or less. Because the ranks rather than the absolute value of the regulatory potential is used, the lower the value the higher the potential. Then the groups of targets can be compared to each other or to a theoretical distribution.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># Figure 2</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>

<span class="co"># plot distance by score of associate peaks</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">distance</span>, <span class="va">ap</span><span class="op">$</span><span class="va">YY1</span><span class="op">$</span><span class="va">peak_score</span>,
     xlab <span class="op">=</span> <span class="st">'Distance'</span>, ylab <span class="op">=</span> <span class="st">'Peak Score'</span>,
     main <span class="op">=</span> <span class="st">'(A)'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="va">ap</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">distance</span>, <span class="va">ap</span><span class="op">$</span><span class="va">YY2</span><span class="op">$</span><span class="va">peak_score</span><span class="op">)</span>

<span class="co"># make labels, colors and groups</span>
<span class="va">labs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Down'</span>, <span class="st">'None'</span>, <span class="st">'Up'</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'green'</span>, <span class="st">'gray'</span>, <span class="st">'red'</span><span class="op">)</span>

<span class="co"># make three groups by quantiles      </span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">dt</span>,<span class="op">~</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">stat</span>, breaks <span class="op">=</span> <span class="fl">3</span>, labels <span class="op">=</span> <span class="va">labs</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># plot the group functions</span>
<span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">pmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">dt</span>, <span class="va">groups</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'(B)'</span>, <span class="st">'(C)'</span><span class="op">)</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">z</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/pkg/target/man/plot_predictions.html">plot_predictions</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">score_rank</span>,
                       group <span class="op">=</span> <span class="va">y</span>, colors <span class="op">=</span> <span class="va">cols</span>, labels <span class="op">=</span> <span class="va">labs</span>,
                       xlab <span class="op">=</span> <span class="st">'Regulatory Potential'</span>, ylab <span class="op">=</span> <span class="st">'ECDF'</span><span class="op">)</span>
      <span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span>
    <span class="op">}</span><span class="op">)</span></pre></div>
<div class="figure">
<img src="workshop_code_files/figure-html/functions-1.png" alt="Predicted functions of YY1 and YY2 on their specific targets. Bindings peaks of the transcription factors in HeLa cells were determined using ChIP-Seq. Distances from the transcription start sites and the transformed distances of the (A) YY1 and YY2 peaks are shown. The regulatory potential of each gene was calculated using target. Genes were grouped into up, none or down regulated based on the fold-change. The emperical cumulative distribution functions (ECDF) of the groups of (C) YY1 and (D) YY2 targets are shown at each regulatory potential value." width="100%"><p class="caption">
Predicted functions of YY1 and YY2 on their specific targets. Bindings peaks of the transcription factors in HeLa cells were determined using ChIP-Seq. Distances from the transcription start sites and the transformed distances of the (A) YY1 and YY2 peaks are shown. The regulatory potential of each gene was calculated using target. Genes were grouped into up, none or down regulated based on the fold-change. The emperical cumulative distribution functions (ECDF) of the groups of (C) YY1 and (D) YY2 targets are shown at each regulatory potential value.
</p>
</div>
<p>The scores of the individual peaks are a decreasing function of the distance from the transcription start sites. The closer the factor binding site from the start site the lower the score. The distribution of these scores is very similar for both factors (Figure @ref(fig:functions)A). The ECDF of the down-regulated of YY1 is higher than that of up- and none-regulated targets (Figure @ref(fig:functions)B). Therefore, the absence of YY1 on its targets result in aggregate in their down regulation. If indeed these are true targets then we expect YY1 to induce their expression. The opposite is true for YY2 where more high ranking targets are up-regulated by the factor knockdown (Figure @ref(fig:functions)C).</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># Table 2</span>
<span class="co"># test individual factor functions</span>
<span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">dt</span>, <span class="va">groups</span>,
     <span class="op">~</span><span class="fu"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">rank</span>,
                       group <span class="op">=</span> <span class="va">.y</span>,
                       compare <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Down'</span>, <span class="st">'Up'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></pre></div>
<table class="table">
<caption>(#tab:tests) Testing for statistical significance of the regulated gene groups.</caption>
<thead><tr class="header">
<th>Factor</th>
<th>Statistic</th>
<th>P.value</th>
<th>Method</th>
<th>Alternative</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>YY1</td>
<td>0.224</td>
<td>2.2e-16</td>
<td>Two-sample KS test</td>
<td>two-sided</td>
</tr>
<tr class="even">
<td>YY2</td>
<td>0.149</td>
<td>2.5e-15</td>
<td>Two-sample KS test</td>
<td>two-sided</td>
</tr>
</tbody>
</table>
<p>To formally test these observations, we use the Kolmogorov-Smirnov (KS) test. The distribution of the two groups are compared for equality. If one lies one either side of the other then they must be drawn from different distributions. Here, we compared the up and down regulated functions for both factors (Table @ref(tab:tests)). In both cases, the distribution of the two groups were significantly different from one another.</p>
</div>
<div id="predicting-the-shared-targets-of-two-factors" class="section level2">
<h2 class="hasAnchor">
<a href="#predicting-the-shared-targets-of-two-factors" class="anchor"></a>Predicting the shared targets of two factors</h2>
<p>Using target to predict the shared target genes and the combined function of the two factors is a variation of the previous analysis. First, the shared/common peaks are generated using the overlap of their genomic coordinates, <code>subsetByOverlaps</code>. Second, Instead of one, two columns for the differential expression statistics one for each factor are needed; these are supplied to the argument <code>stats_col</code> in the same way. Here, <code>common_peaks</code> and <code>both_regions</code> are the main inputs for the analysis functions.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co"># merge and name peaks</span>
<span class="va">common_peaks</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">$</span><span class="va">YY1</span>, <span class="va">peaks</span><span class="op">$</span><span class="va">YY2</span><span class="op">)</span><span class="op">)</span>
<span class="va">common_peaks</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">'common_peak_'</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">common_peaks</span><span class="op">)</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># bind express tables into one</span>
<span class="va">both_express</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">express</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html">nest</a></span><span class="op">(</span><span class="va">fc</span>, <span class="va">pvalue</span>, .key <span class="op">=</span> <span class="st">'values_col'</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/spread.html">spread</a></span><span class="op">(</span><span class="va">tf</span>, <span class="va">values_col</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span><span class="op">(</span><span class="va">YY1</span>, <span class="va">YY2</span>, names_sep <span class="op">=</span> <span class="st">'_'</span><span class="op">)</span>

<span class="co"># make regions using genome and expression data of both factors</span>
<span class="va">both_regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">genome</span>, <span class="va">symbol_entrez</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">both_express</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html">makeGRangesFromDataFrame</a></span><span class="op">(</span>keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="co"># get associated peaks with both factors</span>
<span class="va">common_ap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/target/man/associated_peaks.html">associated_peaks</a></span><span class="op">(</span>peaks <span class="op">=</span> <span class="va">common_peaks</span>,
                              regions <span class="op">=</span> <span class="va">both_regions</span>,
                              regions_col <span class="op">=</span> <span class="st">'tx_id'</span><span class="op">)</span>

<span class="co"># get direct targets of both factors</span>
<span class="va">common_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/target/man/direct_targets.html">direct_targets</a></span><span class="op">(</span>peaks <span class="op">=</span> <span class="va">common_peaks</span>,
                            regions <span class="op">=</span> <span class="va">both_regions</span>,
                            regions_col <span class="op">=</span> <span class="st">'tx_id'</span>,
                            stats_col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'YY1_fc'</span>, <span class="st">'YY2_fc'</span><span class="op">)</span><span class="op">)</span></pre></div>
<p>The output of <code>associated_peaks</code> is the same as before. <code>direct_targets</code> is the same but the <code>stat</code> and the <code>stat_rank</code> carry the product of the two statistics provided in the previous step and the rank of that product.</p>
<p>The output can also be visualized the same way. The targets are divided into three groups based on the statistics product. When the two statistics agree in the sign, the product is positive. This means the knockdown of either transcription factor results in same direction change in the target gene expression. Therefore, the two factors would cooperate if they bind to the same site on that gene. The reverse is true for targets with opposite signed statistics. On these targets, the two factors would be expected to compete for inducing opposing changes in the expression.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Figure 3</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="co"># plot distiace by score for associated peaks</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">common_ap</span><span class="op">$</span><span class="va">distance</span>,
     <span class="va">common_ap</span><span class="op">$</span><span class="va">peak_score</span>,
     xlab <span class="op">=</span> <span class="st">'Distance'</span>,
     ylab <span class="op">=</span> <span class="st">'Peak Score'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(A)'</span><span class="op">)</span>

<span class="co"># make labels, colors and gorups</span>
<span class="va">labs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Competitive'</span>, <span class="st">'None'</span>, <span class="st">'Cooperative'</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'green'</span>, <span class="st">'gray'</span>, <span class="st">'red'</span><span class="op">)</span>

<span class="co"># make three groups by quantiles      </span>
<span class="va">common_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">stat</span>,
                     breaks <span class="op">=</span> <span class="fl">3</span>,
                     labels <span class="op">=</span> <span class="va">labs</span><span class="op">)</span>

<span class="co"># plot predicted function</span>
<span class="fu"><a href="https://rdrr.io/pkg/target/man/plot_predictions.html">plot_predictions</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">score_rank</span>,
                 group <span class="op">=</span> <span class="va">common_groups</span>,
                 colors <span class="op">=</span> <span class="va">cols</span>, labels <span class="op">=</span> <span class="va">labs</span>,
                 xlab <span class="op">=</span> <span class="st">'Regulatory Interaction'</span>, ylab <span class="op">=</span> <span class="st">'ECDF'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">'(B)'</span><span class="op">)</span></pre></div>
<div class="figure">
<img src="workshop_code_files/figure-html/function-1.png" alt="Predicted function of YY1 and YY2 on their shared targets. Shared bindings sites of YY1 and YY2 in HeLa cells were determined using the overlap of the individual factor ChIP-Seq peaks. (A) Distances from the transcription start sites and the transformed distances of the shared peaks are shown. The regulatory interaction of each gene was calculated using target. Genes were grouped into cooperatively, none or competitively regulated based on the the product of the fold-changes from YY1- and YY2-knockdown. (B) The emperical cumulative distribution functions (ECDF) of the groups of targets are shown at each regulatory potential value." width="100%"><p class="caption">
Predicted function of YY1 and YY2 on their shared targets. Shared bindings sites of YY1 and YY2 in HeLa cells were determined using the overlap of the individual factor ChIP-Seq peaks. (A) Distances from the transcription start sites and the transformed distances of the shared peaks are shown. The regulatory interaction of each gene was calculated using target. Genes were grouped into cooperatively, none or competitively regulated based on the the product of the fold-changes from YY1- and YY2-knockdown. (B) The emperical cumulative distribution functions (ECDF) of the groups of targets are shown at each regulatory potential value.
</p>
</div>
<p>The common peaks distances and scores take the same shape (Figure @ref(fig:function)A). The two factors seem to cooperate on more of the common target than any of the two other possibilities (Figure @ref(fig:function)B). This observation can be tested using the KS test. The curve of the cooperative targets lies above that of none and competitively regulated targets (Table @ref(tab:test)).</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># Table 3</span>
<span class="co"># test factors are cooperative</span>
<span class="fu"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">score_rank</span>,
                 group <span class="op">=</span> <span class="va">common_groups</span>,
                 compare <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Cooperative'</span>, <span class="st">'None'</span><span class="op">)</span>,
                 alternative <span class="op">=</span> <span class="st">'greater'</span><span class="op">)</span>

<span class="co"># test factors are more cooperative than competitive</span>
<span class="fu"><a href="https://rdrr.io/pkg/target/man/test_predictions.html">test_predictions</a></span><span class="op">(</span><span class="va">common_dt</span><span class="op">$</span><span class="va">score_rank</span>,
                 group <span class="op">=</span> <span class="va">common_groups</span>,
                 compare <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Cooperative'</span>, <span class="st">'Competitive'</span><span class="op">)</span>,
                 alternative <span class="op">=</span> <span class="st">'greater'</span><span class="op">)</span></pre></div>
<table class="table">
<caption>(#tab:test) Testing for statistical significance of combined functions of the two factors.</caption>
<thead><tr class="header">
<th>Compare</th>
<th>Statistic</th>
<th>P.value</th>
<th>Method</th>
<th>Alternative</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Coop vs None</td>
<td>0.168</td>
<td>1.5e-30</td>
<td>KS test</td>
<td>CDF of x lies above that of y</td>
</tr>
<tr class="even">
<td>Coop vs Comp</td>
<td>0.151</td>
<td>2.2e-16</td>
<td>KS test</td>
<td>CDF of x lies above that of y</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="summary" class="section level1">
<h1 class="hasAnchor">
<a href="#summary" class="anchor"></a>Summary</h1>
<p>In this article, we present a workflow for predicting the direct targets of a transcription factor by integrating binding and expression data. The target package implements the BETA algorithm ranking gene targets based on the distance of the ChIP peaks of the transcription factor in the genes and the differential expression of the factor perturbation. To predict the combined function of two factors, two sets of data are used to find the shared peaks and the product of their differential expression.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Chen2010">
<p>Chen, Li, Toshi Shioda, Kathryn R. Coser, Mary C. Lynch, Chuanwei Yang, and Emmett V. Schmidt. 2010. “Genome-wide analysis of YY2 versus YY1 target genes.” <em>Nucleic Acids Research</em> 38 (12): 4011–26. <a href="https://doi.org/10.1093/nar/gkq112">https://doi.org/10.1093/nar/gkq112</a>.</p>
</div>
<div id="ref-Michaud2013">
<p>Michaud, Joëlle, Viviane Praz, Nicole James Faresse, Courtney K Jnbaptiste, Shweta Tyagi, Frédéric Schütz, and Winship Herr. 2013. “HCFC1 is a common component of active human CpG-island promoters and coincides with ZNF143, THAP11, YY1, and GABP transcription factor occupancy.” <em>Genome Research</em> 23 (6): 907–16. <a href="https://doi.org/10.1101/gr.150078.112">https://doi.org/10.1101/gr.150078.112</a>.</p>
</div>
<div id="ref-Wu2017d">
<p>Wu, Xiao-Nan, Tao-Tao Shi, Yao-Hui He, Fei-Fei Wang, Rui Sang, Jian-Cheng Ding, Wen-Juan Zhang, et al. 2017. “Methylation of transcription factor YY2 regulates its transcriptional activity and cell proliferation.” <em>Cell Discovery</em> 3: 17035. <a href="https://doi.org/10.1038/celldisc.2017.35">https://doi.org/10.1038/celldisc.2017.35</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:mahshaaban@gnu.ac.kr" class="email">mahshaaban@gnu.ac.kr</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mahmoud Ahmed.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
